lib/dao.js

```diff
@@ -2224,11 +2224,24 @@ DataAccessObject.destroyAll = function destroyAll(where, options, cb) {
     }
   }
 
+  if (typeof where === 'object' && Object.keys(where).includes('where')) {
+  /** Option 1
+   *  throw new Error('This method does not accept filter objects. Please pass
+  in a where object');
+  */
+
+  /** Option 2 */
+    where = where.where;
+  }
+
+  if (typeof where === 'object' && Object.keys(where).length === 0) {
+    throw new Error('Cannot pass in an empty where object \'{}\'. Please pass in \'undefined\' instead.');
+  }
+
   cb = cb || utils.createPromiseCallback();
-  where = where || {};
   options = options || {};
 
-  assert(typeof where === 'object', 'The where argument must be an object');
+  // assert(typeof where === 'object', 'The where argument must be an object');
```

Comment:
I believe you will still need checks for the type of the value of where, since you can now pass through null values and such.